generatorreceiver/internal/flags/flag.go

```diff
@@ -139,3 +147,23 @@ func (f *Flag) parent() *Flag {
 	}
 	return Manager.GetFlag(f.cfg.Incident.ParentFlag)
 }
+
+func (ic IncidentConfig) validate() error {
+	if Manager.GetFlag(ic.ParentFlag) == nil {
+		return fmt.Errorf("parent flag %s does not exist", ic.ParentFlag)
+	}
+	if len(ic.Start) == 0 {
+		return fmt.Errorf("start cannot be empty")
+	}
+	if ic.Duration == 0 && len(ic.Start) > 1 {
+		return fmt.Errorf("if duration is not specified, only one start time is permitted (will last until end of incident)")
+	}
+	previousStart := time.Duration(-1)
```

Comment:
This makes me think we might want to ensure that all durations  are >= 0! The check below will handle that for this field but there are quite a few others. Not something to handle now, but I'll add something to the backlog.

Reply:
Yes nice catch